store.go

```diff
@@ -827,36 +839,90 @@ func (s *store) GraphDriver() (drivers.Driver, error) {
 // LayerStore obtains and returns a handle to the writeable layer store object
 // used by the Store.  Accessing this store directly will bypass locking and
 // synchronization, so it is not a part of the exported Store interface.
+// Deprecated: Use RWLayerStore() instead
 func (s *store) LayerStore() (LayerStore, error) {
-	s.graphLock.Lock()
-	defer s.graphLock.Unlock()
```

Comment:
Previously, this lock protected all of `LayerStore`, including `s.getGraphDriver`; now the lock is scoped to `resetIfNeeded`; why is that sufficient? (It might be, but it’s not at all obvious to me — the lock’s scope is not documented.)